packages/atomic/scripts/process-css.mjs

```diff
@@ -82,84 +50,96 @@ const pushImports = (currentFile, importPaths, files) => {
 };
 
 function convertCssToJs(srcPath, distPath, file) {
-  readFile(srcPath, 'utf8', async (err, data) => {
-    if (err) {
-      console.error(chalk.red(`Error reading file: ${srcPath}`));
-      throw err;
-    }
-
-    const files = [file];
-
-    console.log(
-      chalk.blue('Processing:'),
-      chalk.green(`${srcPath} -> ${distPath}`)
-    );
-    const imports = Array.from(data.matchAll(importMatcher)).flatMap(
-      (match) => match
-    );
-    pushImports(srcPath, imports, files);
-    data = data.replace(importWholeLineMatcher, '');
-
-    const result = await processAndMinifyCss(data, srcPath);
-
-    let importIndex = 0;
-    const fileContent = dedent`
-    ${imports.length > 0 ? imports.map((importPath) => `import dep${importIndex++} from '${importPath.replace(/\.css$/, '.css.js')}';`).join('\n') : ''}
-    ${cssToJs(result)}
-    ${
-      imports.length > 0
-        ? `
-    const allCss = [${imports
-      .map((_, index) => `...dep${index}`)
-      .concat('css')
-      .join(', ')}];
-    export default allCss;
-    `
-        : `
-    export default [css];
-    `
-    }
-    `;
-    const jsPath = distPath + '.js';
-
-    writeFile(jsPath, fileContent, (err) => {
-      if (err) {
-        console.error(chalk.red(`Error writing file: ${jsPath}`));
-        throw err;
+  return new Promise(async (resolve, reject) => {
+    try {
+      const data = readFileSync(srcPath, 'utf8');
+      const files = [file];
+
+      console.log(
+        chalk.blue('Processing:'),
+        chalk.green(`${srcPath} -> ${distPath}`)
+      );
+
+      const imports = Array.from(data.matchAll(importMatcher)).flatMap(
+        (match) => match
+      );
+      pushImports(srcPath, imports, files);
+      const cleanedData = data.replace(importWholeLineMatcher, '');
+      const result = await processAndMinifyCss(cleanedData, srcPath);
+
+      let importIndex = 0;
+      const fileContent = dedent`
```

Comment:
Let's refactor this into a function, even if we do need to do string concatenation, so it is more readable (and thus maintainable)